### ISSUE TITLE
Nakamoto Threat Model

### ISSUE THREAD

**Issue by AshtonStephens** [stacks-network/stacks-blockchain#4008]
### Description

**This ticket is to create a threat model for Nakamoto based on the information in the Draft SIP here: https://github.com/stacksgov/sips/pull/155.** The expected outcome of this ticket is a document outlining a concrete threat model for Nakamoto that we can use to aid in designing a secure system from the very beginning.

### Context

A threat model is a systematic process of identifying and analyzing potential threats to a system which will allow us to design and implement security controls to mitigate those threats. It is an important part of the design process for any cryptocurrency, as it helps to ensure that the system is secure and resistant to attack.

**As we move forward with Nakamoto's design and implementation we'll need to identify risk factors and mitigation vectors so we can build in security as we go.**

### Expected Approach

To generate the threat model we must follow the guidance in the following resources:
1. [OWASP Threat Modeling Process](https://owasp.org/www-community/Threat_Modeling_Process)
2. [Threat Modeling: Four Question Framework](https://owasp.org/www-community/Threat_Modeling)

These resources must be the foundation of the research and documentation approach. The final document should include text, diagrams, and any other resources that best articulate the model.

- [OWASP's diagram tool](https://github.com/OWASP/threat-dragon)

Once we have a nearly complete version of this document we'll need to get it reviewed and altered before finally pushing the document to the repository documents.

### Artifacts / Acceptance Criteria

Key stakeholders include (but are not limited to) @wileyj @jcnelson @AshtonStephens

- [ ] Threat model rough draft reviewed by key stakeholders
- [ ] Subsequent iterations of that draft until approved by all key stakeholders
- [ ] An approved and merged document to a permanent resting place for the threat model

**Comment by jo-tm** [comment]
Gotcha

**Comment by AshtonStephens** [comment]
@jo-tm I saw you created a document earlier; can you link that in this ticket?

**Comment by jo-tm** [comment]
https://docs.google.com/document/d/1mAIROw4ExpGVFKp161cBZI-Bm3YYoI96lAKLXEkRKfQ/edit

**Comment by AshtonStephens** [comment]
> > https://docs.google.com/document/d/1mAIROw4ExpGVFKp161cBZI-Bm3YYoI96lAKLXEkRKfQ/edit
> 
> any chance this can be converted into a discussion item in the repo, or at the very least make the doc viewable by others?

Agreed, @jo-tm can you turn this into a discussion? We need to make sure outside stakeholders can see this work.

**Comment by jo-tm** [comment]
@wileyj @AshtonStephens yep I give access to TM, I can give further access to stakeholders from another org.

**Comment by jo-tm** [comment]
Discussion: https://github.com/stacks-network/stacks-core/discussions/4050
